src/pydvl/influence/general.py

```diff
@@ -64,12 +63,17 @@ def compute_influence_factors(
     :returns: An array of size (N, D) containing the influence factors for each
         dimension (D) and test sample (N).
     """
-    test_grads = []
-    for x_test, y_test in maybe_progress(
-        test_data, progress, desc="Batch Test Gradients"
+    test_grads = zero_tensor(
+        shape=(len(test_data.dataset), model.num_params),
+        dtype=test_data.dataset[0][0].dtype,
+    )
```

Comment:
I found there is no device specified, the test_grads tensor is created on CPU while test_data could be on cuda.

Reply:
Also: cuda OOM error occurred here